game-core/src/main/java/games/strategy/triplea/delegate/battle/steps/fire/battlegroup/NormalBattleGroups.java

```diff
@@ -0,0 +1,159 @@
+package games.strategy.triplea.delegate.battle.steps.fire.battlegroup;
+
+import static games.strategy.triplea.delegate.battle.BattleState.Side.DEFENSE;
+import static games.strategy.triplea.delegate.battle.BattleState.Side.OFFENSE;
+import static games.strategy.triplea.delegate.battle.BattleStepStrings.FIRST_STRIKE_UNITS;
+import static games.strategy.triplea.delegate.battle.BattleStepStrings.UNITS;
+
+import games.strategy.engine.data.Unit;
+import games.strategy.engine.data.UnitType;
+import games.strategy.engine.data.properties.GameProperties;
+import games.strategy.triplea.Properties;
+import games.strategy.triplea.delegate.Matches;
+import games.strategy.triplea.delegate.battle.BattleState;
+import java.util.Collection;
+import java.util.List;
+import java.util.function.Predicate;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+import lombok.experimental.UtilityClass;
+import org.triplea.java.PredicateBuilder;
+
+/**
+ * Creates battle groups for units that use attack/defense attributes.
+ *
+ * <p>There is a FirstStrike battle group and a General battle group.
+ */
+@UtilityClass
+class NormalBattleGroups {
+
+  /**
+   * Creates battle groups for the first strike phase
+   *
+   * <p>If an opposing isDestroyer unit is present, then the phase for that side will not occur (see
+   * {@link #noDestroyerPresent} for both firing squadrons)
+   */
+  static BattleGroup createFirstStrike(
+      final Collection<UnitType> unitTypes, final GameProperties properties) {
+
+    final List<FiringSquadron> firingSquadrons =
+        FiringSquadron.createWithTargetInformation(unitTypes).stream()
+            .map(
+                firingSquadron -> {
+                  return createFiringSquadron(
+                          firingSquadron, unitIsFirstStrike(properties), noDestroyerPresent())
+                      .name(FIRST_STRIKE_UNITS)
+                      .build();
+                })
+            .collect(Collectors.toList());
+
+    final BattleGroup.BattleGroupBuilder battleGroupBuilder =
+        BattleGroup.builder()
+            .offenseSquadrons(firingSquadrons)
+            .fireType(BattleState.FireType.NORMAL)
+            .casualtiesOnDefenseReturnFire(false);
+
+    if (Properties.getDefendingSubsSneakAttack(properties)) {
```

Comment:
Can this be inlined with ternaries or defined before hand as variables so that the intermediate builder variable is not needed.

Reply:
See https://github.com/triplea-game/triplea/pull/8602/commits/bb56e98c44c6efaec443b627729f2627d646961f

Reply:
Would be nice to see the commits split out a bit more, it's somewhat tough to go through a large diff to look for just a few lines that have changed.